packages/google-oauth/google_sign-in.js

```diff
@@ -0,0 +1,84 @@
+var Google = require("./namespace.js");
+
+var gplusPromise = new Promise(function (resolve, reject) {
+  if (! Meteor.isCordova) {
+    reject(new Error("plugins.googleplus requires Cordova"));
+    return;
+  }
+
+  document.addEventListener("deviceready", function () {
+    var plugins = global.plugins;
+    var gplus = plugins && plugins.googleplus;
+    if (gplus) {
+      resolve(gplus);
+    } else {
+      reject(new Error("plugins.googleplus not defined"));
+    }
+  }, false);
+});
+
+function tolerateUnhandledRejection() {}
+gplusPromise.catch(tolerateUnhandledRejection);
+
+// After 20 April 2017, Google OAuth login will no longer work from a
+// WebView, so Cordova apps must use Google Sign-In instead.
+// https://github.com/meteor/meteor/issues/8253
+exports.signIn = Google.signIn = function (options, callback) {
+  // support a callback without options
+  if (! callback && typeof options === "function") {
+    callback = options;
+    options = null;
+  }
+
+  gplusPromise.then(function (gplus) {
+    var config = ServiceConfiguration.configurations.findOne({
+      service: "google"
+    });
+
+    if (! config) {
+      throw new ServiceConfiguration.ConfigError();
+    }
+
+    options = Object.assign(Object.create(null), options);
+
+    gplus.login({
+      scopes: getScopes(options).join(" "),
+      webClientId: config.clientId,
+      offline: true
+    }, function (response) {
+      Accounts.callLoginMethod({
+        methodArguments: [Object.assign({
+          googleSignIn: true
+        }, response)],
+        userCallback: callback
+      });
+    }, callback);
+
+  }).catch(callback);
+};
+
+function getScopes(options) {
+  // we need the email scope to get user id from google.
+  var requiredScopes = ['email'];
+  var scopes = ['profile'];
+  if (options && options.requestPermissions) {
+    scopes = options.requestPermissions;
```

Comment:
This isn't the fault of this PR, as this logic existed [before in `google_client.js`](https://github.com/meteor/meteor/pull/8549/files#diff-d3e3aa5afbdb786a7d4e241986164ee6R30), but it's super unclear to me what the difference is between `scope`, `permissions`, and `fields` (and I guess Google [refers to](https://developers.google.com/identity/protocols/OpenIDConnect#userinfocall) the resulting fields as `claims`?)

I digress, is this supposed to be getting checked against [`Google.whitelistedFields`](https://github.com/meteor/meteor/blob/devel/packages/google-oauth/google_server.js#L4)?